pkg/crawl/golang/crawl_test.go

```diff
@@ -35,9 +36,25 @@ func TestCrawler_DetectSrc(t *testing.T) {
 			mockServer: func(w http.ResponseWriter, r *http.Request) {
 				assert.Fail(t, "unexpected HTTP call")
 			},
-			want:    "git::https://github.com/example/repo.git?depth=1",
+			want:    "https://github.com/example/repo",
 			wantErr: false,
 		},
+		{
+			name: "GitHub repository with subdirs",
+			pkg: config.Package{
+				PURL: packageurl.PackageURL{
+					Type:    "golang",
+					Name:    "github.com/example/repo",
+					Subpath: "foo/bar",
+				},
+			},
+			mockServer: func(w http.ResponseWriter, r *http.Request) {
+				assert.Fail(t, "unexpected HTTP call")
+			},
+			want:        "https://github.com/example/repo",
+			wantSubDirs: "/foo/bar",
```

Comment:
The `subdirs` field for `golang` has `/` prefix. But when we take `subdirs` from `url` - `sibdirs` doesn't have this prefix:
https://github.com/aquasecurity/vexhub-crawler/blob/be35d2984258817edaac31653388f67261ddf79f/pkg/url/url_test.go#L34

Currently it doesn't matter because we use `filepath.Join` for this field:
https://github.com/aquasecurity/vexhub-crawler/blob/22835848605ca1c573609b9d367383b3ebbbbc66/pkg/crawl/vex/crawl.go#L65

But maybe we want to take this point into account right away

Reply:
That's a benefit of using `filepath.Join`, so I don't think it's a big deal. However, we might face a problem in the future. Trimmed https://github.com/aquasecurity/vexhub-crawler/pull/28/commits/cdfb58fe803ce5fc545dd418342b974f41ca8796